crates/contract/src/lib.rs

```diff
@@ -2836,4 +2842,98 @@ mod tests {
         let result = contract.migration_info();
         assert_eq!(result, expected_vals);
     }
+
+    fn propose_and_vote(
+        contract: &mut MpcContract,
+        update: Update,
+        expected_update_id: u64,
+    ) -> Vec<dtos::AccountId> {
+        let update_id = contract.proposed_updates.propose(update.clone());
+        assert_eq!(update_id.0, expected_update_id);
+        // generate two accounts for voting
+        let account_id_0 = gen_account_id();
+        let account_id_1 = gen_account_id();
+        contract
+            .proposed_updates
+            .vote(&update_id, account_id_0.clone())
+            .unwrap();
+        contract
+            .proposed_updates
+            .vote(&update_id, account_id_1.clone())
+            .unwrap();
+
+        let mut expected_votes = vec![
+            dtos::AccountId(account_id_0.to_string()),
+            dtos::AccountId(account_id_1.to_string()),
+        ];
+        expected_votes.sort();
+        expected_votes
+    }
+
+    fn test_proposed_updates_case(protocol_contract_state: ProtocolContractState) {
+        let mut contract = MpcContract::new_from_protocol_sate(protocol_contract_state);
+
+        assert_eq!(contract.proposed_updates(), dtos::ProposedUpdates(vec![]));
+
+        let code_update = {
+            // propose update
+            let code = [0u8; 1000];
+            let hash = Sha256::digest(code);
+            let update = Update::Contract(code.into());
+            let expected_update_hash = dtos::UpdateHash::Code(hash.into());
```

Comment:
orthogonal to this PR, but it would probably be much nicer if `Update::Contract` had a digest method instead of this, so that the code needed would be:
```suggestion
            let code = [0u8; 1000];
            let update = Update::Contract(code.into());
            let expected_update_hash = update.digest();
```

Reply:
nice idea!